File: src/FrontEnd/snowy-dune/src/app/models/new-user.ts
export class NewUser {
    id: number;
    firstName: string;
    lastName: string;
    username: string;
    email: string;
    password: string;
    address: string;
    newsletter: number;
    isEnterprise: number;
    phone: string
    authorities: string[];

    constructor(firstName: string,
        lastName: string,
        username: string,
        email: string,
        password: string,
        address: string,
        newsletter: number,
        isEnterprise: number,
        phone: string) {
            this.firstName = firstName;
            this.lastName = lastName;
            this.username = username;
            this.email = email;
            this.password = password;
            this.address = address;
            this.newsletter = newsletter;
            this.isEnterprise = isEnterprise;
            this.phone = phone
        }

}

File: src/BackEnd/SnowyDune/src/main/java/org/iesalixar/poriad/security/dto/NewUser.java
package org.iesalixar.poriad.security.dto;

import java.util.HashSet;
import java.util.Set;

import javax.persistence.Column;
import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;

import org.iesalixar.poriad.entity.Cart;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.sun.istack.NotNull;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class NewUser {
	
	private String firstName;
	
	private String lastName;
	
	@NotBlank
	private String username;
	
	@Email
	private String email;
	
	@NotBlank
	private String password;
	
	private String address;
	
	private String phone;
	
	private boolean newsletter;
	
	private Set<String> roles = new HashSet<>();
	
	
	
}
